woocommerce/woocommerce-gateway-stripe
# Better control of enqueuing new react settings JS files

# Changes proposed in this Pull Request:
Check `$current_tab`, `$current_section` to make sure we are not loading Settings' js files everywhere.

Fixes https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1783

Description: 
- Added a function `WC_Stripe_Helper::should_enqueue_in_current_tab_section()` to check rather we should enqueue a js file or not, based on the given tab and section.
- Removed `is_admin()` check since `admin_enqueue_scripts` hook implies admin pages.
- `upe_settings.js` now only loads inside `wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe`

# Testing instructions
1. Go to all payment methods page `wp-admin/admin.php?page=wc-settings&tab=checkout`
2. Verify `upe_settings.js` is not loaded. 
![image](https://user-images.githubusercontent.com/572862/136465750-b97664d9-cbf0-4511-9b0e-d202a476ad76.png)

3. Go to Stripe payment method `wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe` 
4. Verify `upe_settings.js` is loaded.
![image](https://user-images.githubusercontent.com/572862/136465826-4af2f779-63ee-4be6-863f-4c8976600e37.png)
